build-tasks/generate-recommendations.js

```diff
@@ -8,30 +8,41 @@ const { getAllRules, getMetadataFromFile, getMetadataValue } = require('./common
 const groupedRows = {};
 const warnings = [];
 
-getAllRules().forEach(ruleFile => {
-    const metadata = getMetadataFromFile(ruleFile);
-
-    const groupName = getMetadataValue(metadata, 'group');
-    if (groupName === 'Ignored') {
-        return;
-    }
-    if (groupName === '') {
-        warnings.push('Could not generate recommendation for rule file: ' + ruleFile);
-    }
-    if (groupedRows[groupName] === undefined) {
-        groupedRows[groupName] = [];
-    }
-
-    let recommendation = getMetadataValue(metadata, 'recommendation', true, true);
-    if (recommendation === '') {
-        recommendation = 'true,';
-    }
-    // Replace double quotes with single quote
-    recommendation = recommendation.replace(/"/g, "'");
-
-    const ruleName = getMetadataValue(metadata, 'ruleName');
-    groupedRows[groupName].push(`        '${ruleName}': ${recommendation}`);
-});
+/**
+ * @see https://github.com/Microsoft/tslint-microsoft-contrib/issues/694
+ */
+const ignoredRulesNotYetAdded = new Set(
+    ['banTsIgnore', 'commentType', 'noDefaultImport', 'unnecessaryConstructor'].map(
+        ruleName => `node_modules/tslint/lib/rules/${ruleName}Rule.js`
+    )
+);
+
+getAllRules()
+    .filter(ruleFilePath => !ignoredRulesNotYetAdded.has(ruleFilePath))
+    .forEach(ruleFile => {
+        const metadata = getMetadataFromFile(ruleFile);
+
+        const groupName = getMetadataValue(metadata, 'group');
+        if (groupName === 'Ignored') {
```

Comment:
For simplicity this condition can be extended to also check if `ruleName` (need to be moved from line 43 up) is in `ignoredRulesNotYetAdded` that can be set of rule names instead of file paths.

Will remove `.filter` on all rules and `.map` for set creation.